src/features/Folding.ts

```diff
@@ -101,10 +101,13 @@ class LineNumberRange {
      */
     public rangeKind: vscode.FoldingRangeKind;
 
+    private settings: Settings.ISettings;
+
     constructor(
         rangeKind: vscode.FoldingRangeKind,
     ) {
         this.rangeKind = rangeKind;
+        this.settings = Settings.load();
```

Comment:
Is there a way to ensure that settings are only loaded once? Like passing them into the constructor? Or does `Settings.load()` guarantee that (I assume it does actually, but want to check)?